include/dali/core/cuda_utils.h

```diff
@@ -21,20 +21,18 @@
 #include "dali/core/host_dev.h"
 #include "dali/core/dynlink_cuda.h"
 #include "dali/core/cuda_error.h"
-
-// For the CPU we use half_float lib and float16_cpu type
-namespace half_float {
-
-class half;
-
-}
+#ifndef __CUDA_ARCH__
```

Comment:
perhaps float16 deserves its own header file separated from cuda utils?

Reply:
It does, but maybe let's make it another PR and move it to core then.